src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -32,6 +33,22 @@
 
   private DatastoreService datastore;
 
+  /** 
+   *  Checks if a user was already stored, done when a user logs in
+   * 
+   *  @return 0 on success, -1 when user was not found
```

Comment:
I'm not sure about this method signature.  It would make more sense to me to either

private boolean userExists(int hash), or private Entity fetchUser(string name) or some such, rather than use an int.

Reply:
Ah right, I've been programming a lot in C and forgot about booleans. Will update that.